packages/perseus-editor/src/hint-editor.tsx

```diff
@@ -363,40 +363,29 @@ class CombinedHintsEditor extends React.Component<CombinedHintsEditorProps> {
         ]);
         this.props.onChange({hints: hints}, () => {
             const i = hints.length - 1;
-            // eslint-disable-next-line react/no-string-refs
-            // @ts-expect-error - TS2339 - Property 'focus' does not exist on type 'ReactInstance'.
-            this.refs["hintEditor" + i].focus();
+            this.hintEditors[i]?.focus();
         });
     };
 
     getSaveWarnings: () => any = () => {
-        return _.chain(this.props.hints)
+        return this.props.hints
             .map((hint, i) => {
-                return _.map(
-                    // eslint-disable-next-line react/no-string-refs
-                    // @ts-expect-error - TS2339 - Property 'getSaveWarnings' does not exist on type 'ReactInstance'.
-                    this.refs["hintEditor" + i].getSaveWarnings(),
-                    (issue) => "Hint " + (i + 1) + ": " + issue,
-                );
+                return this.hintEditors["hintEditor" + i]
```

Comment:
I don't think this will work if `this.hintEditors` is a `Record<number, ...>`. It should probably be:

```suggestion
                return this.hintEditors[i]
```